src/config/configureStore.ts

```diff
@@ -1,13 +1,62 @@
+import { List, Record } from 'immutable';
 import { applyMiddleware, combineReducers, createStore } from 'redux';
 import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
+import { createTransform, persistReducer, persistStore } from 'redux-persist';
+import storage from 'redux-persist/lib/storage';
 import promiseMiddleware from 'redux-promise-middleware';
 import thunk from 'redux-thunk';
 import * as ducks from '../ducks';
 
+import { BerthsState, FormsState } from '../types/ducks';
+
+const BerthsTransform = createTransform(
+  (inboundState: BerthsState) => {
+    const { selectedServices, selectedBerths } = inboundState.toObject();
+
+    return {
+      selectedServices: selectedServices.toObject(),
+      selectedBerths: selectedBerths.toArray()
+    };
+  },
+  outboundState => {
+    const berths = Record({
+      selectedServices: Record(outboundState.selectedServices)(),
+      selectedBerths: List(outboundState.selectedBerths)
+    });
+    return berths();
+  },
+  { whitelist: ['berths'] }
+);
+
+const FormsTransform = createTransform(
+  (inboundState: FormsState) => {
+    const forms = inboundState.toObject();
+
+    return forms;
+  },
+  outboundState => {
+    const forms = Record(outboundState);
+    return forms();
+  },
+  { whitelist: ['forms'] }
+);
+
+const persistConfig = {
+  storage,
+  key: 'root',
+  transforms: [BerthsTransform, FormsTransform]
```

Comment:
We should only persist wizard form data?

Reply:
Because every step has its own route and if the user refreshes the page, he'd lose form data from previous steps.